packit/command_handler.py

```diff
@@ -127,26 +153,9 @@ def run_command(
         :param cwd: working directory to run command in
         :param print_live: not used here
         """
-        # we import here so that packit does not depend on sandcastle (and thus python-kube)
-        from sandcastle.api import Sandcastle, MappedDir
-
-        md = MappedDir(
-            local_dir=self.local_project.working_dir,
-            path=cwd or self.config.command_handler_work_dir,
-            with_interim_pvc=True,
-        )
-        sandcastle = Sandcastle(
-            image_reference=self.config.command_handler_image_reference,
-            k8s_namespace_name=self.config.command_handler_k8s_namespace,
-            mapped_dir=md,
-            env_vars=env,
-        )
-        sandcastle.run()
-        try:
-            logger.info(f"Running command: {' '.join(command)}")
-            out: str = sandcastle.exec(command=command)
-        finally:
-            sandcastle.delete_pod()
+        self.run_if_needed()
+        logger.info(f"Running command: {' '.join(command)}")
+        out: str = self.sandcastle.exec(command=command, env=env, cwd=cwd)
```

Comment:
why `self.run_if_needed()` cannot be part of `self.sandcastle.exec` method (e.g check variable if not set, just start it) ?

Reply:
mainly to decouple the code: have smaller methods with clear functionality and call them b/w each other rather than have 2 with all the code

Reply:
I though about simplification like this pseudo code:

   ```
    @property
    @lru_cache
    def sandcastle(self):
        if not self.is_running:
            self.is_running = True
            return Sandcastle_init().run()
```
But that''s true I do not see all consequences of this change, so may lead that my note is completely out of this.

> mainly to decouple the code: have smaller methods with clear functionality and call them b/w each other rather than have 2 with all the code

I though about simplification like this pseudo code:

   ```
    @property
    @lru_cache
    def sandcastle(self):
        if not self.is_running:
            self.is_running = True
            return Sandcastle_init().run()
```
But that''s true I do not see all consequences of this change, so may lead that my note is completely out of this.

Reply:
makes sense, fixed